discovery/azure/azure.go

```diff
@@ -348,10 +360,13 @@ func (d *Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
 
 			// Get the IP address information via separate call to the network provider.
 			for _, nicID := range vm.NetworkInterfaces {
-				networkInterface, err := client.getNetworkInterfaceByID(ctx, nicID)
-				if err != nil {
-					level.Error(d.logger).Log("msg", "Unable to get network interface", "name", nicID, "err", err)
-					ch <- target{labelSet: nil, err: err}
+				networkInterface, detailedErr := client.getNetworkInterfaceByID(ctx, nicID)
+				if detailedErr != nil {
+					if detailedErr.StatusCode == 404 {
```

Comment:
If you need to access elements of a specific error type, use `errors.As`.

Reply:
Fixed.